user_guide_src/source/incoming/auto_routing_improved.rst

```diff
@@ -0,0 +1,488 @@
+methods.
+
+.. important:: For security reasons, if a controller is used in the defined routes,
+    Auto Routing (Improved) does not route to the controller.
+
+.. note:: Auto Routing (Improved) is disabled by default. To use it, see
+    :ref:`enabled-auto-routing-improved`.
+
+**************************************
+Differences from Auto Routing (Legacy)
+**************************************
+
+:ref:`auto-routing-legacy` is a routing system from CodeIgniter 3. If you are not
+familiar with it, go to the next section.
+
+If you know it well, these are some changes in **Auto Routing (Improved)**:
+
+- A controller method needs HTTP verb prefix like ``getIndex()``, ``postCreate()``.
+    - Since developers always know the HTTP method, a request with an unexpected
+      HTTP method will never execute the controller.
+- The Default Controller (``Home`` by default) and the Default Method (``index`` by default) must be omitted in the URI.
+    - It restricts one-to-one correspondence between controller methods and URIs.
+    - E.g. by default, you can access ``/``, but ``/home`` and ``/home/index``
+      will be 404 Not Found.
+- It checks method parameter count.
+    - If there are more parameters in the URI than the method parameters, it results
+      in 404 Not Found.
+- It does not support ``_remap()`` method.
+    - It restricts one-to-one correspondence between controller methods and URIs.
+    - But it has the :ref:`auto-routing-improved-default-method-fallback` feature
+      instead.
+- Can't access controllers in Defined Routes.
+    - It completely separates controllers accessible via **Auto Routing** from
+      those accessible via **Defined Routes**.
+
+.. _enabled-auto-routing-improved:
+
+******************************
+Enable Auto Routing (Improved)
+******************************
+
+To use it, you need to change the setting ``$autoRoute`` option to ``true`` in
+**app/Config/Routing.php**::
+
+    public bool $autoRoute = true;
+
+And you need to change the property ``$autoRoutesImproved`` to ``true`` in
+**app/Config/Feature.php**::
+
+    public bool $autoRoutesImproved = true;
+
+.. important:: When you use Auto Routing (Improved), you must remove the line
+    ``$routes->get('/', 'Home::index');``  in **app/Config/Routes.php**. Because
+    defined routes take precedence over Auto Routing, and controllers defined in
+    the defined routes are denied access by Auto Routing (Improved) for security
+    reasons.
+
+************
+URI Segments
+************
+
+The segments in the URL, in following with the Model-View-Controller approach,
+usually represent::
+
+    http://example.com/{class}/{method}/{param1}
+
+1. The first segment represents the controller **class** that should be invoked.
+2. The second segment represents the class **method** that should be called.
+3. The third, and any additional segments, represent any **parameters** that will be passed to the controller.
```

Comment:
I took from this sentence that the parameters are sent to the controller. 
But the parameters are sent to the methods in the controller.

Reply:
Changed to `the controller method`.